### PR TITLE
Analytics background worker - counts & codeowners

### PR DESCRIPTION
**Pull request by cbart**
Part of #52826

This change adds a background worker that indexes total file counts and codeowned file counts for a repo.

## Test plan

Tests with real database and fake gitserver.

### CODE REVIEW

File: enterprise/internal/own/background/analytics.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	Name:      "own_analytics_files_indexed_total",
+})
+
+func (r *analyticsIndexer) indexRepo(ctx context.Context, repoId api.RepoID) error {
+	repoStore := r.db.Repos()
+	repo, err := repoStore.Get(ctx, repoId)
+	if err != nil {
+		return errors.Wrap(err, "repoStore.Get")
+	}
+	files, err := r.client.LsFiles(ctx, nil, repo.Name, "HEAD")
```
Comment (sashaostrikov) [inline enterprise/internal/own/background/analytics.go:54]: we definitely need to check how it will affect gitserver on scaletesting

**Review by sashaostrikov — APPROVED** [review]
Nice! :shipit::shipit::shipit:

**Comment by Strum355** [comment]
70 files changed, rebase mistake? 👀

**Comment by cbart** [comment]
> 70 files changed, rebase mistake? 👀

I have no clue how this happened 🤷

**Pull request merged**